contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -713,48 +716,58 @@ func (s *SchedulerServer) bootstrap(hks hyperkube.Interface, sc *schedcfg.Config
 		return n.(*api.Node)
 	}
 
-	fcfs := scheduler.NewFCFSPodScheduler(as, lookupNode)
-	mesosPodScheduler := scheduler.New(scheduler.Config{
-		Schedcfg:          *sc,
+	fcfs := podschedulers.NewFCFSPodScheduler(as, lookupNode)
+	framework := framework.New(framework.Config{
+		SchedulerConfig:   *sc,
 		Executor:          executor,
-		Scheduler:         fcfs,
 		Client:            client,
-		EtcdClient:        etcdClient,
-		FailoverTimeout:   s.FailoverTimeout,
-		ReconcileInterval: s.ReconcileInterval,
-		ReconcileCooldown: s.ReconcileCooldown,
+		FailoverTimeout:   s.failoverTimeout,
+		ReconcileInterval: s.reconcileInterval,
+		ReconcileCooldown: s.reconcileCooldown,
 		LookupNode:        lookupNode,
+		StoreFrameworkId: func(id string) {
+			// TODO(jdef): port FrameworkId store to generic Kubernetes config store as soon as available
+			_, err := etcdClient.Set(meta.FrameworkIDKey, id, uint64(s.failoverTimeout))
+			if err != nil {
+				log.Errorf("failed to renew frameworkId TTL: %v", err)
+			}
+		},
 	})
 
-	masterUri := s.MesosMaster
+	masterUri := s.mesosMaster
 	info, cred, err := s.buildFrameworkInfo()
 	if err != nil {
 		log.Fatalf("Misconfigured mesos framework: %v", err)
 	}
 
-	schedulerProcess := ha.New(mesosPodScheduler)
+	schedulerProcess := ha.New(framework)
 	dconfig := &bindings.DriverConfig{
 		Scheduler:        schedulerProcess,
 		Framework:        info,
 		Master:           masterUri,
 		Credential:       cred,
-		BindingAddress:   s.Address,
-		BindingPort:      uint16(s.DriverPort),
-		HostnameOverride: s.HostnameOverride,
+		BindingAddress:   s.address,
+		BindingPort:      uint16(s.driverPort),
+		HostnameOverride: s.hostnameOverride,
 		WithAuthContext: func(ctx context.Context) context.Context {
-			ctx = auth.WithLoginProvider(ctx, s.MesosAuthProvider)
-			ctx = sasl.WithBindingAddress(ctx, s.Address)
+			ctx = auth.WithLoginProvider(ctx, s.mesosAuthProvider)
+			ctx = sasl.WithBindingAddress(ctx, s.address)
 			return ctx
 		},
 	}
 
-	kpl := scheduler.NewPlugin(mesosPodScheduler.NewDefaultPluginConfig(schedulerProcess.Terminal(), s.mux))
-	runtime.On(mesosPodScheduler.Registration(), func() { kpl.Run(schedulerProcess.Terminal()) })
-	runtime.On(mesosPodScheduler.Registration(), s.newServiceWriter(schedulerProcess.Terminal()))
+	// create scheduler loop
+	eventBroadcaster := record.NewBroadcaster()
```

Comment:
Hmm, we call `NewBroadcaster()` in two places now, here and in https://github.com/kubernetes/kubernetes/pull/16316/files#diff-ccc01b5b821b0c9fc4cc1a272cc0c276R83. Is this intended?